backend/local/backend_plan.go

```diff
@@ -206,7 +207,12 @@ func (b *Local) renderPlan(plan *plans.Plan, baseState *states.State, schemas *t
 // output values will not currently be rendered because their prior values
 // are currently stored only in the prior state. (see the docstring for
 // func planHasSideEffects for why this is and when that might change)
-func RenderPlan(plan *plans.Plan, baseState *states.State, schemas *terraform.Schemas, ui cli.Ui, colorize *colorstring.Colorize) {
+func RenderPlan(plan *plans.Plan, baseState, priorState *states.State, schemas *terraform.Schemas, ui cli.Ui, colorize *colorstring.Colorize) {
+	if statesShowDrift(baseState, priorState) {
+		// We've detected some drift by refreshing during our planning, then.
+		renderDrift(baseState, priorState, schemas, ui, colorize)
+	}
```

Comment:
I think if we use this approach for rendering drift, there are two non-ideal consequences:

- `terraform plan -out=saved.tfplan` and `terraform show saved.tfplan` will not render the same result (the drift will be dropped)
- `terraform show -json saved.tfplan` does not have enough information to render drift

There are a couple of ways that I can think of handling this (either or both of which may be flawed):

1. Extend the plan file format to store a set of objects similar to `ResourceInstanceChange` for the drifted resources, computing those from the diff between the two states, and then correspondingly update the plan renderer and JSON plan output
2. Store the base state in the plan file along with the refreshed state, use those to render the plan

As a potential consumer of the JSON plan, I would much rather see option 1 if possible, so long as the `resource_drift` objects are semantically equivalent to the existing `resource_changes` objects. I can then reuse the same logic to do whatever I want with that information. If only option 2 is possible, I'd have to write new logic to diff the state files myself, which is possible but feels like more work.